src/main/java/io/vertx/junit5/VertxTestContext.java

```diff
@@ -113,6 +114,11 @@ public synchronized void completeNow() {
    */
   public synchronized void failNow(Throwable t) {
     Objects.requireNonNull(t, "The exception cannot be null");
+
+    // Unsure on this, but don't want to prevent the overuse of a strict checkpoint's flag() failing the test with its own exception
+    if (!(t instanceof IllegalStateException)) {
+      assertContextNotCompletedOrFailed();
+    }
```

Comment:
@jponge - I added this clause as calling flag() more than once on a strict checkpoint fails the context with an IllegalStateException, and I didn't want the new ISE I added to override that one. But it doesn't feel like a great approach.

Reply:
The problem is that you could have an `IllegalStateException` for some other reason than a strict checkpoint failing

Reply:
Good point. What if I subclassed IllegalStateException for `checkpoint.flag()` to throw, so I could special case that precisely? It would also preserve the API if someone is catching ISE around a call to `flag` for some reason.

Reply:
Hi @jponge I've pushed another version up, when you have time to look at it :)